pepper-apis/dsm-core/src/main/java/org/broadinstitute/lddp/email/SendGridClient.java

```diff
@@ -20,29 +20,27 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * Used for integrating with SendGrid...
+ * SendGridClient is a legacy wrapper for using SendGrid, carried over from
+ * the legacy DDP repo, and used exclusively by DSM.
  */
 public class SendGridClient {
 
     private static final Logger logger = LoggerFactory.getLogger(SendGridClient.class);
 
     private static final String LOG_PREFIX = "SendGrid";
+    private final SendGrid sendGrid;
 
-    private String sendGridKey;
-    private JsonObject settings = null;
-
-    public void configure(String sendGridKey, JsonObject settings) {
-        this.sendGridKey = sendGridKey;
-        this.settings = settings;
+    public SendGridClient(SendGrid sendGrid) {
```

Comment:
Instead of passing in key and settings values and mutating state via `configure()`, we just pass in a fully formed `SendGrid` object and then clients pass in `settings` as needed when sending emails.